File: Source/Backup/WebSites/DevelopmentWebsite/Common/JavaScript/CouponsAvailable.js
///Copyright       : Copyright (c) Sopra Group India Ltd 
///Project         : Timeless Tributes
///File Name       : TributePortal.DevelopmentWebsite.Common.JavaScript.CouponsAvailable.js
///Author          : 
///Creation Date   : 
///Description     : This file is used to check for coupons' availability
///Audit Trail     : Date of Modification  Modified By         Description



var xmlHttp
function UserValidate(UserName,Password)
{ 
     xmlHttp=GetXmlHttpObject();
    if (xmlHttp==null)
    {
        alert ("Your browser does not support AJAX!");
        return;
    } 
     var thePage="/DevelopmentWebsite/ModelPopup/CheckAvailabality.aspx";      
           thePage=thePage+"?User="+UserName.value;
           thePage=thePage+"&Pass="+Password.value;                
           
     xmlHttp.onreadystatechange=stateChanged;     
     xmlHttp.open("POST",thePage,false);     
     //xmlHttp.open("GET",thePage,false);     
     xmlHttp.send(null);   
}
function stateChanged() 
{     
    if (xmlHttp.readyState==4)
    {     
      var id=$('txtHint');     
      id.innerHTML=xmlHttp.responseText;  

       if(id.innerText.trim()=='true')
      {     
      var title1 = document.title;       
       if (title1 == "UserRegistration")
            {window.location="UserAccounts.aspx";}
       else
           {window.location = location.href;}
      
       window.reload();
      }
      else
      {		              
       var pass=$('ctl00_txtLoginPassword1');  
       var pass2=$('txtLoginPassword1');  
       if(pass)
       {    
        pass.value="";
        pass.focus();
       }
       if(pass2)
       {    
        pass2.value="";
        pass2.focus();
       }
      }
    }
}
function GetXmlHttpObject()
{
    var xmlHttp=null;
    try
    {
      // Firefox, Opera 8.0+, Safari
      xmlHttp=new XMLHttpRequest();
    }
    catch (e)
    {
      // Internet Explorer
        try
        {
            xmlHttp=new ActiveXObject("Msxml2.XMLHTTP");
        }
            catch (e)
        {
        xmlHttp=new ActiveXObject("Microsoft.XMLHTTP");
        }
    }
    return xmlHttp;
}

File: Source/Backup/BusinessEntities/CouponsAvailable.cs
///Copyright       : Copyright (c) Sopra Group India Ltd 
///Project         : Timeless Tributes
///File Name       : TributesPortal.BusinessEntities.CouponsAvailable.cs
///Author          : 
///Creation Date   : 
///Description     : This file defines the attributes of the Coupons Availability object
///Audit Trail     : Date of Modification  Modified By         Description

using System;

namespace TributesPortal.BusinessEntities
{
    [Serializable]
    public class CouponsAvailable
    {
        /// <summary>
        /// Default Contructor
        /// <summary>
        public CouponsAvailable()
        { }


        public int CouponID
        {
            get { return _CouponID; }
            set { _CouponID = value; }
        }
        private int _CouponID;


        public int PrimaryCouponID
        {
            get { return _PrimaryCouponID; }
            set { _PrimaryCouponID = value; }
        }
        private int _PrimaryCouponID;


        public decimal SerialNumber
        {
            get { return _SerialNumber; }
            set { _SerialNumber = value; }
        }
        private decimal _SerialNumber;


        public string CouponCode
        {
            get { return _CouponCode; }
            set { _CouponCode = value; }
        }
        private string _CouponCode;


        public int Status
        {
            get { return _Status; }
            set { _Status = value; }
        }
        private int _Status;


        public bool IsActive
        {
            get { return _IsActive; }
            set { _IsActive = value; }
        }
        private bool _IsActive;


        public bool IsDelete
        {
            get { return _IsDelete; }
            set { _IsDelete = value; }
        }
        private bool _IsDelete;

        /// <summary>
        /// User defined Contructor
        /// <summary>
        public CouponsAvailable(int CouponID,
            int PrimaryCouponID,
            decimal SerialNumber,
            string CouponCode,
            int Status,
            bool IsActive,
            bool IsDelete)
        {
            this._CouponID = CouponID;
            this._PrimaryCouponID = PrimaryCouponID;
            this._SerialNumber = SerialNumber;
            this._CouponCode = CouponCode;
            this._Status = Status;
            this._IsActive = IsActive;
            this._IsDelete = IsDelete;
        }

    }
}
